### PR TITLE
Fixes a race in deviceplugin/manager_test.go and a race in deviceplug…

### PR DESCRIPTION
**Pull request by jiayingz**
…in/manager.go.

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
https://github.com/kubernetes/kubernetes/issues/52560

**Special notes for your reviewer**:
Tested with  go test -count 50 -race k8s.io/kubernetes/pkg/kubelet/deviceplugin and all runs passed.

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by RenaudWasTaken** [comment]
We are still leaking a goroutine in `device_plugin_stub.go` by not closing the close channel during the `Stop` implementation

File: pkg/kubelet/deviceplugin/manager.go [hunk 1]
```diff
@@ -189,9 +189,12 @@ func (m *ManagerImpl) Register(ctx context.Context,
 
 // Stop is the function that can stop the gRPC server.
 func (m *ManagerImpl) Stop() error {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
```
Comment (RenaudWasTaken) [inline pkg/kubelet/deviceplugin/manager.go]: I'd rather we call Unlock at the end of the for loop :)

**Comment by dims** [comment]
@jiayingz @RenaudWasTaken Do we need this for 1.8?

File: pkg/kubelet/deviceplugin/device_plugin_stub.go [hunk 2]
```diff
@@ -70,7 +70,7 @@ func (m *Stub) Start() error {
 	// Wait till grpc server is ready.
 	for i := 0; i < 10; i++ {
```
Comment (RenaudWasTaken) [inline pkg/kubelet/deviceplugin/device_plugin_stub.go]: This whole loop is useless. `m.server.GetServiceInfo()` returns the number of services as soon as RegisterDevicePluginServer is called

Comment (jiayingz) [reply pkg/kubelet/deviceplugin/device_plugin_stub.go]: I think it returns the number of services when they are ready, not as soon as RegisterDevicePluginServer is called.

Comment (RenaudWasTaken) [reply pkg/kubelet/deviceplugin/device_plugin_stub.go]: https://github.com/grpc/grpc-go/blob/master/server.go#L376

You can also test this

Comment (jiayingz) [reply pkg/kubelet/deviceplugin/device_plugin_stub.go]: Hmm, this is interesting to know. I thought GetServiceInfo returns ready-to-serve no. of methods, but looking at the code, it is indeed should return No. of methods right after registration. However, I did verify that if I removed this code block, running 'bazel test --runs_per_test=20 //pkg/kubelet/deviceplugin:go_default_test' became flaky (I think mostly on endpoint_test.go) but with the current code, the tests all passed. I think we will need to spend more time understand this code. For now, I would like to keep it this way to make sure the tests are not flaky.

**Pull request merged**